src/main/java/nextstep/courses/domain/Session.java

```diff
@@ -0,0 +1,39 @@
+package nextstep.courses.domain;
+
+import nextstep.courses.CannotCreateSessionException;
+import nextstep.courses.CannotRegisterException;
+import nextstep.users.domain.NsStudent;
+import nextstep.users.domain.NsUser;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Session {
+    private Long id;
+    private Period sessionPeriod;
+    private SessionState sessionState;
+    private PayStrategy registerStrategy;
+    private Image coverImage;
+    private List<NsStudent> students = new ArrayList<>();
+
+    public Session(Long id, LocalDateTime startDate, LocalDateTime endDate, SessionState sessionState, PayStrategy registerStrategy, Image coverImage) {
+        this.id = id;
+        this.sessionPeriod = new Period(startDate, endDate);
+        this.sessionState = sessionState;
+        this.registerStrategy = registerStrategy;
+        this.coverImage = coverImage;
+    }
+
+    private void validateDate(LocalDateTime startDate, LocalDateTime endDate) {
+        if (startDate.isAfter(endDate)) {
+            throw new CannotCreateSessionException("강의의 시작 날짜가 끝나는 날짜보다 뒤입니다.");
+        }
+    }
+
+    public NsStudent register(NsUser user, NaturalNumber money) {
+        NsStudent student = Register.registerSession(user, id, money, sessionState, registerStrategy, students);
+        students.add(student);
+        return student;
+    }
```

Comment:
세션이 학생을 만들고 등록에게 넘겨주는게 자연스럽겠군요 🤔

```suggestion
    public NsStudent register(NsUser user, NaturalNumber money) {
        registerStrategy.pay(money);
        NsStudent student = new NsStudent(user.getId(), this.id);
        registry.register(student);
        return student;
    }
```

```java
public class Registry {
    private final int capacity;

    private final List<NsStudent> students;

    private final SessionState sessionState;

    public void register(NsStudent student) {
        if (!sessionState.canNotRegister()) {
            throw new CannotRegisterException("강의는 모집 중일 때만 등록할 수 있습니다.");
        }
        Students students = new Students(this.capacity, this.students);
        students.enroll(student);
    }
}
```

이런식으로 구현할 수 있을것 같아요! (ide 가 아닌곳에서 코드 작성하는게 어렵네요 😅)
이렇게하면 Registry 는 유틸성 클래스가 아닌 하나의 도메인으로 볼 수 있을것도 같습니다.

그리고 Students 에서 추가로 코멘트 드렸던 책임들을 수행하면 될것 같아요!